File: 0078-subsets/0078-subsets.java
class Solution {
    List<List<Integer>> result = new ArrayList<>();
    private void solve(int[]nums,int idx,List<Integer> curr){
        if(idx==nums.length){
            result.add(new ArrayList<>(curr));
            return;
        }
        curr.add(nums[idx]);
        solve(nums,idx+1,curr);
        curr.remove(curr.size()-1);
        solve(nums,idx+1,curr);
    }
    public List<List<Integer>> subsets(int[] nums) {
        
        solve(nums,0,new ArrayList<>());
        return result;
        
    }
}

File: 0078-subsets/0078-subsets.cpp
class Solution {
public:
    vector<vector<int>> result;
    void solve(vector<int> & nums , int idx,vector<int>&curr){
        if(idx==nums.size()){
            result.push_back(curr);
            return;
        }
        
        curr.push_back(nums[idx]);
        solve(nums,idx+1,curr);
        curr.pop_back();
        solve(nums,idx+1,curr);


    }
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<int>curr;
        solve(nums,0,curr);
        return result;

        
    }
};